File: frontend/src/types/agent_message.ts
/**
 * TypeScript types for agent chat messages
 */

export interface FileAttachment {
  filename: string;
  file_path: string;
  file_size: number;
  content_type: string;
}

export interface AgentMessage {
  id: number;
  workspace_task_id: number;
  role: "user" | "assistant" | "system";
  content: string;
  attachments?: FileAttachment[] | null;
  created_at: string;
  user_id: number | null;
  username: string | null;
}

export interface AgentMessageCreate {
  content: string;
}

export interface AgentMessageListResponse {
  messages: AgentMessage[];
  total: number;
}

File: backend/app/schemas/agent_message.py
"""Pydantic schemas for agent message API validation and serialization."""

from datetime import datetime

from pydantic import BaseModel, Field


# Request schema for sending a message
class AgentMessageCreate(BaseModel):
    """Schema for creating a new agent message."""

    content: str = Field(..., min_length=1, max_length=10000, description="Message content")


# File attachment schema
class FileAttachment(BaseModel):
    """Schema for file attachment metadata."""

    filename: str
    file_path: str
    file_size: int
    content_type: str


# Response schema for a message
class AgentMessageResponse(BaseModel):
    """Schema for agent message in responses."""

    id: int
    workspace_task_id: int
    role: str  # "user", "assistant", "system"
    content: str
    attachments: list[FileAttachment] | None = None
    created_at: datetime
    user_id: int | None
    username: str | None  # Denormalized for display

    class Config:
        from_attributes = True


# List response
class AgentMessageListResponse(BaseModel):
    """Schema for agent message list endpoint response."""

    messages: list[AgentMessageResponse]
    total: int
